File: JavaScript/7js 1/hasUniqueChars.js
/*
|--------------------------------------------------------------------------
| All unique
|--------------------------------------------------------------------------
|
*/

function hasUniqueChars(str) {
   return str == [...new Set(str)].join("");
}


function hasUniqueChars(str) {
   return !/(.).*\1/.test(str);
}


function hasUniqueChars(str) {
   return new Set(str).size == str.length;
}


console.log(hasUniqueChars("  nAa"), false);
console.log(hasUniqueChars("abcdef"), true);
console.log(hasUniqueChars("aA"), true);
console.log(hasUniqueChars("++-"), false);

File: Python/7py 2/has_unique_chars.py
"""
|--------------------------------------------------------------------------
| All unique
|--------------------------------------------------------------------------
|
"""

def has_unique_chars(string):
   unq = []
   [unq.append(i) for i in string if i not in unq]
   return string == ''.join(unq)


def has_unique_chars(string):
   return len(string) == len(set(string))


def has_unique_chars(string):
   return list(dict.fromkeys(string)) == list(string)


print(has_unique_chars("abcdef"), True)
print(has_unique_chars("++-"), False)
print(has_unique_chars("  nAa"), False)